### DISCUSSION TITLE
How to show loading/progress ring when video is being buffered/downloading from url

### DISCUSSION THREAD

**Discussion by insomniachi** [SuRGeoNix/Flyleaf#429] [Q&A]
How to show loading/progress ring when video is being buffered/downloading from url

**Comment by SuRGeoNix** [comment]
Hi @insomniachi,

Buffering progress and downloading are different things. For player to check when it buffers you can use the events BufferingStarted/BufferingCompleted (percentage wouldn't be accurate here but you could check the demuxer's buffered duration compared to min buffered duration of the config) for downloader the property change for Downloader.DownloadPercentage (you can also check the downloader sample).

Let me know if that works for you.

**Reply by insomniachi** [reply]
when internet speed is low, there will be a progress ring in sites like youtube.
should I use buffering start to show the progress ring and buffering completed to hide it?

**Reply by SuRGeoNix** [reply]
Yes, you could use also percentage based on values that I said
